File: Pages/LoginPage.js
const {BasePage} = require('./BasePage');

class LoginPage extends BasePage {
    constructor(page) {
        super(page)

        this.usernameInput = '#user-name';
        this.passwordInput = '#password';
        this.loginbtn = '#login-button';
        this.errorText = '[data-test = "error"]';
    }
    async open(){
        await this.goto();
    }

    async login(username, password){
        await this.page.fill(this.usernameInput, username);
        await this.page.fill(this.passwordInput, password);
        await this.page.click(this.loginbtn);
    }
    async getErrorMessage(){
        return await this.page.textContent(this.errorText);
    }
}
module.exports = {LoginPage};

File: Pages/LoginPage.cs
using Microsoft.Playwright;
using System.Threading.Tasks;

namespace PlaywrightSpecFlowSauceDemo.Pages
{
    public class LoginPage : BasePage
    {
        // Selectors
        private readonly string _usernameField = "#user-name";
        private readonly string _passwordField = "#password";
        private readonly string _loginButton = "#login-button";
        private readonly string _errorMessage = "[data-test='error']";

        public LoginPage(IPage page) : base(page)
        {
        }

        public async Task NavigateToLoginPage(string baseUrl)
        {
            await NavigateTo(baseUrl);
        }

        public async Task EnterUsername(string username)
        {
            await Type(_usernameField, username);
        }

        public async Task EnterPassword(string password)
        {
            await Type(_passwordField, password);
        }

        public async Task ClickLogin()
        {
            await Click(_loginButton);
        }

        public async Task<bool> IsErrorDisplayed()
        {
            return await IsVisible(_errorMessage);
        }
    }
}
